cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/BasicWorkflowToken.java

```diff
@@ -27,19 +32,139 @@
  */
 public class BasicWorkflowToken implements WorkflowToken {
   private Map<String, Map<String, Long>> mapReduceCounters;
```

Comment:
Why need to keep it separate? Can it be just a filtered map view on the `tokenValueMap` for the `getMapReduceCounters` method?